dao/src/main/java/by/it/novik/dao/PaymentDao.java

```diff
@@ -119,6 +123,10 @@ private Criteria getCriteriaOfFilter(PaymentsFilter paymentsFilter) {
         double maxAmountPayment = paymentsFilter.getMaxAmountPayment();
         Criteria criteria = getSession().createCriteria(Payment.class);
         //To check existing filters
+        // please refactor, this is really hard to understand what us going on here and it would hard to maintain such code
+        // and moreover some conditions would literally never happen, they are unreachable code (thank to my Intellij IDEA)
+        // http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
+        // what about using Restrictions.Conjunction() method?
```

Comment:
I use in this method object Criterion for condition in the query. I have to choose the parameters to obtain the final query to be added to the query. These parameters are the fields of filters and user can choose one, two or no one filter. And I have to check this conditions, but they arn't included in the query. So, I have to use the constructions if...else before making query. I know that it is hard to understand. I can only transfer this conditions in another function.